src/states/elder/mod.rs

```diff
@@ -464,6 +471,18 @@ impl Elder {
                 self.vote_for_event(event.clone());
             });
 
+        let ps = self
+            .set_of_proving_sections
+            .iter()
+            .filter(|(_, si)| our_pfx.is_neighbour(si.prefix()))
```

Comment:
hmm is this correct cos this is only going to be considering the leaf sec_info, not any sec_infos from within the ProvingSection itself? maybe take a look at `handle_proving_sections_event`. That should show how its not just the leaf sec_info we aim to discover about but any within the ProvingSection too.

Reply:
we should also check if the version in this sec_info is relevant as in if we cached v-5 and neighbour_infos has v-6 for the same pfx, they're still our neighbour according to this filter but no point voting for this excess PS here

Reply:
> hmm is this correct cos this is only going to be considering the leaf sec_info, not any sec_infos from within the ProvingSection itself? maybe take a look at handle_proving_sections_event. That should show how its not just the leaf sec_info we aim to discover about but any within the ProvingSection too.

After a discussion we decided that the cache should primarily function on the leaf

Reply:
> hmm is this correct cos this is only going to be considering the leaf sec_info, not any sec_infos from within the ProvingSection itself? maybe take a look at handle_proving_sections_event. That should show how its not just the leaf sec_info we aim to discover about but any within the ProvingSection too.